modules/weapons/src/missiles.rs

```diff
@@ -0,0 +1,787 @@
+use serde::{Deserialize, Serialize};
+use crate::{Damages, Speed, WeaponInformations};
+
+/// The projectile type is the type of trajectory the missile will be using
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, PartialOrd)]
+pub enum ProjectileType {   
+    /// The missile is guided by a human operator
+    ///
+    /// The missile is able to change his trajectory after launch and is able to follow a target.
+    /// He uses a cruise trajectory with a low altitude.
+    Cruiser = 0,
+    /// The missile is guided by a radar
+    ///
+    /// His trajectory is a parabola, and he can go up to 100 km in altitude.
+    /// He can't change his trajectory after launch.
+    Ballistic = 1
+}
+
+impl TryFrom<i64> for ProjectileType {
+    type Error = ();
+
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(ProjectileType::Cruiser),
+            1 => Ok(ProjectileType::Ballistic),
+            _ => Err(())
+        }
+    }
+}
+
+/// The missile guidance type is the type of guidance that is used in the missile
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, PartialOrd)]
+pub enum MissileGuidanceType {
+    /// The missile is guided by a human operator
+    Laser = 0,
+    /// The missile is guided by a radar
+    Radar = 1,
+    /// The missile is guided by a heat source
+    Heat = 2,
+    /// The missile is guided by a GPS signal
+    Gps = 3,
+    /// The missile is guided by a radio signal
+    Radio = 4
+}
+
+impl TryFrom<i64> for MissileGuidanceType {
+    type Error = ();
+
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(MissileGuidanceType::Laser),
+            1 => Ok(MissileGuidanceType::Radar),
+            2 => Ok(MissileGuidanceType::Heat),
+            3 => Ok(MissileGuidanceType::Gps),
+            4 => Ok(MissileGuidanceType::Radio),
+            _ => Err(())
+        }
+    }
+}
+
+/// The warhead type is the type of warhead that is used in the missile
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, PartialOrd)]
+pub enum WarheadType {
+    /// Cruise missile
+    Cruiser = 0,
+    /// Anti-ship missile
+    AntiShip = 1,
+    /// Anti-aircraft missile
+    AntiAircraft = 2,
+    /// Anti-ballistic missile
+    Abm = 3,
+    /// Short-range ballistic missile
+    Srbm = 4,
+    /// Medium-range ballistic missile
+    Mrbm = 5,
+    /// Intercontinental ballistic missile
+    Icbm = 6,
+    /// Electromagnetic pulse missile
+    ///
+    /// # Note
+    /// Should only be used with a nuclear charge, otherwise it's useless and will do nothing
```

Comment:
EMP still used as a complement to nuclear missile or can be used as a warhead on his own

Reply:
Normally an EMP is specific and a nuclear missile cannot be used to do it, that's why I declared it as its own type